internal/access/access_key.go

```diff
@@ -27,9 +34,15 @@ func ListAccessKeys(c *gin.Context, identityID uid.ID, name string, showExpired
 }
 
 func CreateAccessKey(c *gin.Context, accessKey *models.AccessKey) (body string, err error) {
-	db, err := RequireInfraRole(c, models.InfraAdminRole)
-	if err != nil {
-		return "", HandleAuthErr(err, "access key", "create", models.InfraAdminRole)
+	var db data.GormTxn
+
+	if accessKey.IssuedFor == GetRequestContext(c).Authenticated.User.ID {
+		db = getDB(c) // can create access keys for yourself.
```

Comment:
Could a user use an access key to create others indefinitely (effectively nulling the point of having access keys expire)? Just a thought to explore. Perhaps we should limit this somehow to the "first" access key created by the user's user/pass or IDP login?

Reply:
Good call, I added a check for that. There's some additional conditions to explore here, like we might want the option to not let access keys create other access keys in general. We'll have to move to a more fine-grained permission model for that (consuming our own grant system)